core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -596,14 +599,33 @@ protected void doRun() throws Exception {
                 finishBecauseUnavailable(shardId, writeConsistencyFailure);
                 return;
             }
-            final ReplicationPhase replicationPhase;
+            ReplicationPhase replicationPhase = null;
+            DiscoveryNode relocatingNodeId = null;
             try {
-                indexShardReference = getIndexShardOperationsCounter(shardId);
-                Tuple<Response, ReplicaRequest> primaryResponse = shardOperationOnPrimary(state.metaData(), request);
-                if (logger.isTraceEnabled()) {
-                    logger.trace("action [{}] completed on shard [{}] for request [{}] with cluster state version [{}]", transportPrimaryAction, shardId, request, state.version());
+                indexShardReference = getIndexShardOperationsCounter(shardId, true);
```

Comment:
get we call the method getIndexShardReference? Also I like explicit naming here better then a primary boolean. See [seq_no branch](https://github.com/elastic/elasticsearch/blob/feature/seq_no/core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java#L710) for an example